### PR TITLE
SOV-1019: add close LOC popup

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-1019

### CODE REVIEW

File: apps/frontend/src/app/3_organisms/ZeroLocForm/CloseCreditLine.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  availableBalance,
+  onSubmit,
+}) => {
+  const { t } = useTranslation();
+  const [error, setError] = useState(false);
+  const availableTokens = [SupportedTokens.dllr, SupportedTokens.zusd];
+  const tokens = availableTokens.map(token => ({
+    value: token,
+    label: token.toUpperCase(),
+  }));
```
Comment (creed-victor) [inline apps/frontend/src/app/3_organisms/ZeroLocForm/CloseCreditLine.tsx]: Move it outside of component

File: apps/frontend/src/app/3_organisms/ZeroLocForm/CloseCreditLine.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  const insufficientBalance = useMemo(() => {
+    return Number(availableBalance) - Number(creditValue);
+  }, [availableBalance, creditValue]);
+
+  useEffect(() => {
+    Number(availableBalance) < Number(creditValue)
+      ? setError(true)
+      : setError(false);
+  }, [creditValue, availableBalance]);
```
Comment (creed-victor) [inline apps/frontend/src/app/3_organisms/ZeroLocForm/CloseCreditLine.tsx]: having useMemo would be cleaner:

```suggestion
  const hasError = useMemo(() => Number(availableBalance) < Number(creditValue), [creditValue, availableBalance]);
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

File: apps/frontend/src/app/2_molecules/AssetRenderer/AssetRenderer.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    const assetData = SupportedTokenList.find(item => item.symbol === asset);
+    if (!assetData) {
+      throw new Error(`Asset ${asset} not found in the list.`);
+    }
+    return assetData.getIcon();
+  }, []);
+
+  useEffect(() => {
+    const getLogo = async () => {
+      const logo = await getAssetLogo(asset);
```
Comment (tiltom) [inline apps/frontend/src/app/2_molecules/AssetRenderer/AssetRenderer.tsx]: Use `getTokenDetails` getter instead. I know you only need the logo but you can get rid of the logo resolving + the logo will be cached if you use the getter.

Also, fetch an asset logo only if it's needed (`showAssetLogo && !logo`).

File: apps/frontend/src/app/2_molecules/AssetRenderer/AssetRenderer.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+   * Applied classNames to the asset element.
+   * @default false
+   * */
+  assetClassName?: string;
+  /**
+   * Applied classNames to the outer element.
+   * */
+  className?: string;
+  /**
+   * Applied data-layout-id to the outer element.
```
Comment (tiltom) [inline apps/frontend/src/app/2_molecules/AssetRenderer/AssetRenderer.tsx]:
```suggestion
   * Applied data attribute to the outer element.
```

File: apps/frontend/src/app/2_molecules/AssetRenderer/AssetRenderer.module.css [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.container {
```
Comment (tiltom) [inline apps/frontend/src/app/2_molecules/AssetRenderer/AssetRenderer.module.css:1]: There is `1rem` margin between `AssetRenderer` and the select button in the design but only `0.5rem` in the implementation. I think you need to add `0.5rem` to either `.container` or `.asset`.

Design:
image

Implementation:
image

File: apps/frontend/src/app/3_organisms/ZeroLocForm/CloseCreditLine.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  );
+
+  return (
+    <>
+      <Paragraph
+        className="font-medium"
+        size={ParagraphSize.base}
+        children={t(translations.closeCreditLine.title)}
+        dataLayoutId="close-credit-line"
+      />
```
Comment (tiltom) [inline apps/frontend/src/app/3_organisms/ZeroLocForm/CloseCreditLine.tsx]:
```suggestion
           <Paragraph
        className="font-medium"
        size={ParagraphSize.base}
        children={t(translations.closeCreditLine.title)}
        dataLayoutId="close-credit-line"
        style={ParagraphStyle.tall}
      />
```
Line height should be `1.25rem` according to the design.

File: apps/frontend/src/app/3_organisms/ZeroLocForm/utils.ts [hunk 7]
```diff
@@ -9,3 +11,9 @@ export const normalizeAmountByType = (
   }
   return Math.abs(amount) * -1;
 };
+
+export const availableTokens = (tokens: SupportedTokens[]) =>
```
Comment (tiltom) [inline apps/frontend/src/app/3_organisms/ZeroLocForm/utils.ts]:
```suggestion
export const tokensToOptions = (tokens: SupportedTokens[]) =>
```

File: apps/frontend/src/app/5_pages/ZeroPage/ZeroPage.tsx [hunk 8]
```diff
@@ -46,6 +49,17 @@ export const ZeroPage: FC = () => {
...
   }, [account, liquity]);
 
+  useEffect(() => {
+    const getZUSDBalance = async () => {
+      const balance = (await liquity.getZUSDBalance(account)).toString();
+      return balance;
+    };
+
+    if (account && liquity) {
+      getZUSDBalance().then(balance => setZusdBalance(balance));
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/ZeroPage/ZeroPage.tsx]:
```suggestion
      getZUSDBalance().then(setZusdBalance);
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

**Review by LunaSov — APPROVED** [review]

**Review by LunaSov — APPROVED** [review]

**Pull request merged**